test/e2e/framework/ssh/ssh.go

```diff
@@ -135,11 +135,27 @@ func NodeSSHHosts(c clientset.Interface) ([]string, error) {
 
 	sshHosts := make([]string, 0, len(hosts))
 	for _, h := range hosts {
-		sshHosts = append(sshHosts, net.JoinHostPort(h, SSHPort))
+		if canConnect(h) {
```

Comment:
I'm a bit worried about this behavior when there are many nodes with no SSH: it will try all nodes sequentially, each time timing out after 3 seconds. That can become quite slow overall.

Perhaps all nodes should be checked in parallel?

Reply:
Alright, changed the loop to be run in parallel. :+1: